web/src/app/forms/FormContainer.js

```diff
@@ -1,97 +1,58 @@
-import React from 'react'
-import p from 'prop-types'
+import React, { useEffect, useState } from 'react'
+import PropTypes from 'prop-types'
 import MountWatcher from '../util/MountWatcher'
 
 import { FormContext, FormContainerContext } from './context'
 import { get, set, cloneDeep } from 'lodash'
 
 // FormContainer handles grouping multiple FormFields.
 // It works with the Form component to handle validation.
-export class FormContainer extends React.PureComponent {
-  static propTypes = {
-    value: p.object,
-
-    errors: p.arrayOf(
-      p.shape({
-        message: p.string.isRequired,
-        field: p.string.isRequired,
-        helpLink: p.string,
-      }),
-    ),
-
-    onChange: p.func,
-    disabled: p.bool,
-
-    mapValue: p.func,
-    mapOnChangeValue: p.func,
-
-    // If true, will render optional fields with `(optional)` appended to the label.
-    // In addition, required fields will not be appended with `*`.
-    optionalLabels: p.bool,
-
-    // Enables functionality to remove an incoming value at it's index from
-    // an array field if the new value is falsey.
-    removeFalseyIdxs: p.bool,
-  }
-
-  static defaultProps = {
-    errors: [],
-    value: {},
-    onChange: () => {},
-
-    mapValue: (value) => value,
-    mapOnChangeValue: (value) => value,
-  }
-
-  state = {
-    validationErrors: [],
-  }
 
+export function FormContainer(props) {
+  const setState = useRef()
+  // const [state, setState] = useState({})
   _fields = {}
-
-  addField = (fieldName, validate) => {
-    if (!this._fields[fieldName]) {
-      this._fields[fieldName] = []
+  function addField(fieldName, validate) {
+    if (_fields[fieldName]) {
+      _fields[fieldName] = []
```

Comment:
`useRef` works a little differently, we can initialize it to an empty object and then reference it via the `.current` property. We can simplify it since we never update the base object.

```js
  const _fields = useRef({}).current

  function addField(fieldName, validate) {
    if (_fields[fieldName) {
      _fields[fieldName] = []
```